cdap-ui/app/cdap/components/AbstractWidget/InputKeyValueDropdown/index.tsx

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as React from 'react';
+
+import AbstractMultiRowWidget, {
+  IMultiRowProps,
+  IMultiRowWidgetProps,
+} from 'components/AbstractWidget/AbstractMultiRowWidget';
+import { IStageSchema } from 'components/AbstractWidget';
+import KeyValueDropdownRow, {
+  IDropdownOption,
+  OrderingEnum,
+} from 'components/AbstractWidget/KeyValueDropdownWidget/KeyValueDropdownRow';
+
+import ThemeWrapper from 'components/ThemeWrapper';
+import { WIDGET_PROPTYPES } from 'components/AbstractWidget/constants';
+import { objectQuery } from 'services/helpers';
+
+interface IInputKeyValueDropdownWidgetProps extends IMultiRowWidgetProps {
+  'key-placeholder'?: string;
+  'kv-delimiter'?: string;
+  dropdownOptions: IDropdownOption[];
+  delimiter?: string;
+  showDelimiter?: boolean;
+  extraConfig?: any;
+}
+
+interface IInputKeyValueDropdownProps extends IMultiRowProps<IInputKeyValueDropdownWidgetProps> {}
+
+interface IField {
+  name: string;
+  type: string[];
+}
+
+class InputKeyValueDropdownWidgetView extends AbstractMultiRowWidget<IInputKeyValueDropdownProps> {
```

Comment:
Can we rename this as `InputFieldDropdownKeyValue`? We already have `InputFieldDropdown` and `KeyValueDropdown`. This is basically a merge between them.